File: Networking/Objects/NetworkObjectSyncMessage.cs
using Networking.Data;

using System;
using System.Collections.Generic;

namespace Networking.Objects
{
    public struct NetworkObjectSyncMessage : ISerialize, IDeserialize
    {
        public List<Type> syncTypes;

        public NetworkObjectSyncMessage(List<Type> syncTypes)
            => this.syncTypes = syncTypes;

        public void Deserialize(Reader reader)
        {
            syncTypes = reader.ReadList<Type>(() => reader.ReadType());
        }

        public void Serialize(Writer writer)
        {
            writer.WriteList(syncTypes, type => writer.WriteType(type));
        }
    }
}

File: Networking/Objects/NetworkObjectSyncMessage.vb
Imports Networking.Data

Imports System
Imports System.Collections.Generic

Namespace Networking.Objects
    Public Structure NetworkObjectSyncMessage
        Implements ISerialize, IDeserialize
        Public syncTypes As Dictionary(Of Short, Type)

        Public Sub New(syncTypes As Dictionary(Of Short, Type))
            Me.syncTypes = syncTypes
        End Sub

        Public Sub Deserialize(reader As Reader) Implements IDeserialize.Deserialize
            syncTypes = reader.ReadDictionary(Of Short, Type)()
        End Sub

        Public Sub Serialize(writer As Writer) Implements ISerialize.Serialize
            writer.WriteDictionary(syncTypes)
        End Sub
    End Structure
End Namespace
